free-programming-books.md

```diff
@@ -379,6 +380,7 @@
 
 * [Competitive Programmer's Handbook](https://cses.fi/book.html) - Antti Laaksonen (PDF)
 * [Competitive Programming, 1st Edition](https://cpbook.net/#CP1details) (PDF)
+* [Competitive Programming, 2nd Edition](https://www.comp.nus.edu.sg/~stevenha/myteaching/competitive_programming/cp2.pdf) - S Halim, F Halim (PDF)
```

Comment:
also does not appear to be an authorized distribution

Reply:
This is an authorized distribution, the second version of the book is now retired and the author made it available in the website cpbook.net, it's easy to check... I just used a different link ...

Reply:
This is an authorized distribution, the second version of the book is now retired and the author made it available in the website cpbook.net, it's easy to check... I just used a different link ...